ftwhttp/header.go

```diff
@@ -35,9 +35,9 @@ func (w stringWriter) WriteString(s string) (n int, err error) {
 
 // Add adds the key, value pair to the header.
 // It appends to any existing values associated with key.
-// The key is case in-sensitive
+// The key is case-insensitive
 func (h Header) Add(key, value string) {
-	key = canonicalHeaderKey(key)
+	key = canonicalKey(key)
```

Comment:
Sorry, I just realised that using `canonicalKey` doesn't work. We need to retain the original casing of the header name. Instead, `Get` should use `canonicalKey` when comparing keys for retrieval.

Reply:
I do not understand how it would work if only `Get` retrieves data by the "canonicalized" key. The original [header.go](https://cs.opensource.google/go/go/+/refs/tags/go1.22.5:src/net/http/header.go;l=30-58) works in a different way if we want to model it after them

Reply:
There are at least two possible ways to accomplish this:
1. iterate over the keys of the map, canonicalise each key and compare it to the canonicalised version of the search key
2. modify the internal state, e.g., use a map `(canonical(key)) -> (map(header-name, header-value))`, then you can use an efficient lookup to find all the headers that reduce to the canonical form